docs/chainlink-keepers/register-upkeep.md

```diff
@@ -35,7 +35,7 @@ After you register your contract as an Upkeep on the Keepers Registry, the Keepe
 
      - Provide the **Upkeep address** of the contract you want to automate. The contract doesn't need to be validated on-chain, but for Keepers to work it needs to be [Keepers-compatible](../compatible-contracts/).
      - Enter an **email** to receive Upkeep notifications. The email address will be encrypted.
-     - The **Gas Limit** is the maximum amount of gas that will be used to execute your function on-chain. The Gas Limit of the [KeepersCounters.sol](/docs/chainlink-keepers/compatible-contracts#example-contract) example contract should be set to 200,000. We simulate `performUpkeep` during `checkUpkeep` and if the gas exceeds this limit the function will not execute on-chain. The gas limit cannot exceed the `callGasLimit` in the [configuration of the registry](/docs/chainlink-keepers/supported-networks/#configurations).
+     - The **Gas Limit** is the maximum amount of gas that will be used to execute your function on-chain. The Gas Limit of the [KeepersCounters.sol](/docs/chainlink-keepers/compatible-contracts#example-contract) example contract should be set to 200,000. We simulate `performUpkeep` during `checkUpkeep` and if the gas exceeds this limit the function will not execute on-chain. The gas limit cannot exceed the `callGasLimit` in the [configuration of the registry](/docs/chainlink-keepers/supported-networks/#configurations). *Note*: When your Upkeep is eligible for execution, the Chainlink Keepers Network will first simulate your `performUpkeep` and only execute an on-chain transaction if the `performUpkeep` does not revert. This will protect your Upkeep against possible logical errors in your contracts.
```

Comment:
There are actually 2 different gas limits, callGasLimit: limit for checkUpkeep and performGasLimit: limit for perform. 
The line "We simulate `performUpkeep` during `checkUpkeep` and if the gas exceeds this limit the function will not execute on-chain. The gas limit cannot exceed the `callGasLimit` in the [configuration of the registry](/docs/chainlink-keepers/supported-networks/#configurations)" is a bit confusing in this aspect, could we clarify this

Reply:
Also it is no longer exactly true that we simulate `performUpkeep` during `checkUpkeep`. (We do simulate performUpkeep before broadcasting a tx, but it's independent of checkUpkeep)

Reply:
We also mention it [here](https://documentation-git-fork-ritika-b-keepers-v2-chainlinklabs.vercel.app/docs/chainlink-keepers/compatible-contracts/#performupkeep-function)

Reply:
resolved